### ISSUE TITLE
[v2] UnusedImport false positive for companion object member extension properties

### ISSUE THREAD

**Issue by DanielSouzaBertoldi** [detekt/detekt#9333]
## Expected Behavior
When a companion object member extension property is imported and used, `UnusedImport` should not report the import as unused. The code compiles correctly and the import is required.

## Observed Behavior
`UnusedImport` flags imports of the form `ClassName.Companion.extensionProperty` as unused, even when the property is actively used in the same file.

Curiously, this false positive **only occurs with `detektMain`**. Running `detekt` correctly passes with no findings.

### Reproducer
Just [clone the reproducer here](https://github.com/DanielSouzaBertoldi/detekt-unused-import-false-positive) and run `./gradlew detektMain`. Detekt should throw:

```
e: /project-dir/Handler.kt:4:1 The import 'org.example.Keybindings.Companion.isNotTriggered' is unused. [UnusedImport]
e: /project-dir/Handler.kt:5:1 The import 'org.example.Keybindings.Companion.isTriggered' is unused. [UnusedImport]
```

### Manually reproducing the false positive

1. Create a new project
2. Add detekt 2.0.0-alpha.3
3. Add these two files:

```kotlin
// **`src/main/kotlin/org/example/Keybindings.kt`**
package org.example

data class Event(val key: String)

open class Keybindings {
      companion object : Keybindings()

      val Event.isTriggered: Boolean get() = key == "A"
      val Event.isNotTriggered: Boolean get() = key != "A"
}

// **`src/main/kotlin/org/example/handler/Handler.kt`**
package org.example.handler

import org.example.Event
import org.example.Keybindings.Companion.isNotTriggered
import org.example.Keybindings.Companion.isTriggered

fun handle(event: Event) = event.isTriggered || event.isNotTriggered 
```

4. Now, run `./gradlew detektMain`. It should throw this error:

```
e: Handler.kt:4:1 The import 'org.example.Keybindings.Companion.isTriggered' is unused. [UnusedImport]
e: Handler.kt:5:1 The import 'org.example.Keybindings.Companion.isNotTriggered' is unused. [UnusedImport]
```

## Context
We hit this [while migrating from detekt v1 to v2](https://github.com/JetBrains/intellij-community/pull/3495#discussion_r3163724201). Our codebase uses companion object member extensions for keybinding definitions (extension properties on `KeyEvent` declared in an interface, implemented by a singleton companion). With v2's type resolution, every one of these imports gets flagged, so we've had to add `@file:Suppress("UnusedImport")` to affected files as a workaround.

## Your Environment

* Version of detekt used: 2.0.0-alpha.3
* Version of Gradle used (if applicable): 9.3.0
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: MacOS Tahoe 26.2
* Link to your project (if it's a public repository): https://github.com/DanielSouzaBertoldi/detekt-unused-import-false-positive

**Comment by nchen000** [comment]
I am interested in picking this up. I will open an PR once I have a fix ready.
